components/admin_console/billing/company_info_edit.tsx

```diff
@@ -71,11 +67,15 @@ const CompanyInfoEdit: React.FC<Props> = () => {
 
     useEffect(() => {
         dispatch(getCloudCustomer());
-    }, []);
+    }, [dispatch]);
 
     useEffect(() => {
         setValidation();
-    }, [sameAsBillingAddress, companyName, address, city, postalCode, country, state]);
+    }, [setValidation]);
```

Comment:
I think this one was a warning, but I think it's a valid case. With the previous version, we might forget to call `setValidation` if we added, for example, another field to this page, but by doing it this way, `setValidation` will change when any of the fields it checks change, and that'll force this to run.

Alternatively, I think the better way to fix this would be to just get rid of the `useState` for `isValid` and just recalculate that value on every render. We don't even really need that derived state

Reply:
In this case, the validation is purely based on other values in the state though, so we can derive that at render time because `isValid` will only change if something else in state changes which would already cause an update. Instead of rendering once when, for example, the company name changes and then a second time when `isValid` changes, we could remove the state and only have to render once